File: CryptChat.F#/Services/TelegramBotService.fs
namespace CryptChat.Services

module TelegramBotService =
    open System
    open System.Net.Http
    open System.Threading
    open System.Threading.Tasks
    open Telegram.Bot
    open Telegram.Bot.Types
    open CryptChat.Base

    let botToken = "8130511485:AAHeCpnorWzj3ZYJPsDo-TE_bGncdqiJlTk"
    let botClient = TelegramBotClient(botToken)

    let setWebhookAsync (webhookUrl: string) =
        task {
            use client = new HttpClient()
            let! _ = client.PostAsync(webhookUrl, null)
            Console.WriteLine("Webhook set successfully!")
        }

    let handleMessageAsync (message: Message) (logic: Logic) (cancellationToken: CancellationToken) : Task =
        task {
            let chatId = message.Chat.Id
            let userId = message.From.Id
            let mutable isAdmin = false

            match message.Text with
            | text when text.StartsWith("/") -> 
                let! admins = botClient.GetChatAdministrators(ChatId.op_Implicit chatId)
                isAdmin <- admins |> Seq.exists (fun admin -> admin.User.Id = userId)
            | _ -> ()

            match message.Text with
            | text when text.StartsWith("/start") -> do! logic.StartAsync(message, botClient)
            | text when text.StartsWith("/lastBlock") -> 
                let responseMessage = if isAdmin then "Вы админ" else "Вы не админ"
                do! botClient.SendMessage(ChatId.op_Implicit chatId, responseMessage, cancellationToken = cancellationToken) :> Task
            | _ -> return ()
        }

File: CryptChat/Services/TelegramBotService.cs
using Telegram.Bot.Types;
using Telegram.Bot;
using CryptChat.Base;

namespace CryptChat.Services
{
    public class TelegramBotService
    {
        private readonly TelegramBotClient _botClient;

        public TelegramBotService(string botToken)
        {
            _botClient = new TelegramBotClient(botToken);
        }

        public async Task SetWebhookAsync(string webhookUrl)
        {
            var client = new HttpClient();
            var response = await client.PostAsync(webhookUrl, null);
            Console.WriteLine("Webhook set successfully!");
        }

        public async Task HandleMessageAsync(Message message, Logic logic, CancellationToken cancellationToken)
        {
            var chatId = message.Chat.Id;
            var userId = message.From!.Id;

            bool isAdmin = false;

            if (message.Text != null)
            {
                if (message.Text.StartsWith("/"))
                {
                    isAdmin = await CheckIfUserIsAdmin(chatId, userId);
                }

                if (message.Text == "/start")
                {
                    await logic.StartAsync(message, _botClient);
                }

                if (message.Text == "/lastBlock")
                {
                    var responseMessage = isAdmin ? "Вы админ" : "Вы не админ";
                    await _botClient.SendMessage(chatId, responseMessage, cancellationToken: cancellationToken);
                }
            }
        }

        private async Task<bool> CheckIfUserIsAdmin(long chatId, long userId)
        {
            try
            {
                var administrators = await _botClient.GetChatAdministrators(new ChatId(chatId));
                return administrators.Any(admin => admin.User.Id == userId);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Ошибка при проверке админов: {ex.Message}");
                return false;
            }
        }

        /*// Обработка callback-кнопок
async Task HandleCallbackQuery(ITelegramBotClient botClient, CallbackQuery callbackQuery)
{
    await botClient.SendMessage(callbackQuery.Id, "Обработан callback!");
}*/
    }
}
